R/run-nmtran.R

```diff
@@ -0,0 +1,375 @@
+  if(is.na(status_val)){
+    cli::cli_abort("NM-TRAN terminated unexpectedly")
+  }else if(status_val == 0){
+    status <- "Finished Running"
+  }else{
+    status <- "Failed. See errors."
+  }
+
+  # Tabulate NM-TRAN results
+  nmtran_results <- list(
+    nmtran_model = basename(nmtran_input),
+    run_dir = run_dir,
+    status = status,
+    status_val = status_val,
+    output = nmtran.p$stdout
+  )
+
+  return(nmtran_results)
+}
+
+
+#' Helper for finding and reading in a `bbi.yaml` file given a `bbi_nonmem_model`
+#' object or explicit `.config_path`.
+#' @inheritParams run_nmtran
+#' @noRd
+get_bbi_yaml_path <- function(.mod = NULL, .config_path = NULL){
+  if(!is.null(.mod)){
+    check_model_object(.mod, "bbi_nonmem_model")
+    model_dir <- get_model_working_directory(.mod)
+  }
+  bbi_yaml_path <- .config_path %||% file.path(model_dir, "bbi.yaml")
+
+  if(!file_exists(bbi_yaml_path)){
+    if(is.null(.config_path)){
+      cli::cli_abort(
+        c(
+          "No bbi configuration was found in the execution directory.",
+          "i" = "Please run {.fun bbr::bbi_init} with the appropriate directory to continue."
+        )
+      )
+    }else{
+      cli::cli_abort("No bbi configuration was found at {.path {(.config_path)}}")
+    }
+  }
+
+  return(bbi_yaml_path)
+}
+
+
+#' Parse `.bbi_args` and return the three expected `nmfe_options` for `NM-TRAN`
+#'  in the correct format
+#' @inheritParams run_nmtran
+#' @param nmfe_options named list of nmfe options defined in a `bbi.yaml` file.
+#'
+#' @details
+#'
+#' Combines NONMEM submission args and consolidates to `NMFE` arguments only
+#'  - The arguments of interest are `prdefault`, `tprdefault`, and `maxlim`,
+#'  which impact the evaluation of `NM-TRAN`
+#'  - Priority: `.bbi_args` > model yaml > `bbi.yaml`
+#'
+#' @examples
+#'
+#' \dontrun{
+#'
+#'  bbi_yaml_path <- get_bbi_yaml_path(.mod)
+#'  bbi_yaml <- yaml::read_yaml(bbi_yaml_path)
+#'
+#'  parse_nmtran_args(
+#'    .mod,
+#'    .bbi_args = list(nm_version = "nm74gf", prdefault = TRUE),
+#'    nmfe_options = bbi_yaml$nmfe_options
+#'  )
+#'
+#' }
+#' @keywords internal
+#' @return a named character vector
+parse_nmtran_args <- function(
+    .mod,
+    .bbi_args = NULL,
+    nmfe_options = NULL
+){
+
+  # These are the default NMFE options that are passed to `NM-TRAN`
+  #  - They should always be passed (in the correct order), so set the
+  #    defaults here and merge with other bbi_args to ensure these are passed
+  #    to NM-TRAN
+  nmfe_args_def <- list(prdefault = FALSE, tprdefault = FALSE, maxlim = 2)
+
+  # Mimic submit_model behavior: a FALSE value for .bbi_arg has no effect
+  # - Filter these out from .bbi_args
+  if(!is.null(.bbi_args)){
+    check_bbi_args(.bbi_args)
+    req_nmtran_args <- paste(names(nmfe_args_def), collapse = "|")
+    .bbi_args <- purrr::imap(.bbi_args, function(val, bbi_arg){
+      if(str_detect(bbi_arg, req_nmtran_args) && isFALSE(val)){
+        return(NULL)
```

Comment:
This filtering should happen after the merge of `bbi_args` from the model YAML.

Reply:
I got a bit turned around trying to make sure I got this right. Im pushing dedicated tests to help clarify the expected result in a number of scenarios. If those all look good I think we'll have this logic down.